src/main/scala/wdl4s/Task.scala

```diff
@@ -16,15 +16,48 @@ import scala.util.Try
 object Task {
   val Ws = Pattern.compile("[\\ \\t]+")
   def apply(ast: Ast, wdlSyntaxErrorFormatter: WdlSyntaxErrorFormatter): Task = {
-    val name = ast.getAttribute("name").asInstanceOf[Terminal].getSourceString
+    val nameTerminal = ast.getAttribute("name").asInstanceOf[Terminal]
+    val name = nameTerminal.getSourceString
     val declarations = ast.findAsts(AstNodeName.Declaration).map(Declaration(_, "name", wdlSyntaxErrorFormatter))
     val commandAsts = ast.findAsts(AstNodeName.Command)
-    if (commandAsts.size != 1) throw new UnsupportedOperationException("Expecting exactly one Command section")
+    if (commandAsts.size != 1) throw new SyntaxError("Expecting exactly one Command section")
     val commandTemplate = commandAsts.head.getAttribute("parts").asInstanceOf[AstList].asScala.toVector map {
       case x: Terminal => new StringCommandPart(x.getSourceString)
       case x: Ast => ParameterCommandPart(x, wdlSyntaxErrorFormatter)
     }
 
+    commandTemplate collect { case x: ParameterCommandPart => x } foreach { p =>
+      p.expression.ast.findTerminals("identifier") foreach {variable =>
+        if (!declarations.map(_.name).contains(variable.getSourceString)) {
+          throw new SyntaxError(wdlSyntaxErrorFormatter.commandExpressionContainsInvalidVariableReference(nameTerminal, variable))
+        }
+      }
+    }
+
+    def findDeclarationAsts(decl: Declaration) =
+      ast.findAsts(AstNodeName.Declaration) collect { case a: Ast if a.getAttribute("name").sourceString == decl.name => a }
+
+    declarations.foldLeft(Seq.empty[Declaration]) { (acc, cur) =>
```

Comment:
This foldLeft function is a little unwieldy in-line. Can you break it out into a separate def?